src/cargo/ops/cargo_doc.rs

```diff
@@ -35,15 +35,16 @@ pub fn doc(ws: &Workspace<'_>, options: &DocOptions) -> CargoResult<()> {
             .join(&name)
             .join("index.html");
         if path.exists() {
+            let config_browser = {
+                let cfg = ws.config().get::<CargoDocConfig>("doc")?;
+
+                cfg.browser
+                    .map(|path_args| (path_args.path.resolve_program(ws.config()), path_args.args))
```

Comment:
I think to remove the warning, the following can be changed to this:

```suggestion
                let cfg: Option<PathAndArgs> = ws.config().get("doc.browser")?;
                cfg.map(|path_args| (path_args.path.resolve_program(ws.config()), path_args.args))
```

and just delete the `CargoDocConfig` struct.  The issue is that the config system is expecting the `doc` config structure to only include the browser field.

Alternatively, `Config.doc_extern_map` could be changed to something that returns a struct that includes both fields (browser and extern-map).